src/abstractdataframe/show.jl

```diff
@@ -458,41 +485,58 @@ function showrows(io::IO,
     return
 end
 
-#' @exported
-#' @description
-#'
-#' Render an AbstractDataFrame to an IO system. The specific visual
-#' representation chosen depends on the width of the REPL window
-#' from which the call to `show` derives. The dynamic response 
-#' to screen width can be configured using the `allcols` argument.
-#'
-#' @param io::IO The IO system to which `df` will be printed.
-#' @param df::AbstractDataFrame An AbstractDataFrame.
-#' @param allcols::Bool Should only a subset of columns that fits
-#'        the device width be printed? Defaults to `false`.
-#' @param rowlabel::Symbol What label should be printed when rendering the
-#'        numeric ID's of each row? Defaults to `"Row"`.
-#' @param displaysummary::Bool Should a brief string summary of the
-#'        AbstractDataFrame be rendered to the IO system before printing the
-#'        contents of the renderable rows? Defaults to `true`.
-#'
-#' @returns `nothing` value.
-#'
-#' @examples
-#'
-#' df = DataFrame(A = 1:3, B = ["x", "y", "z"])
-#' show(stdout, df, false, :Row, true)
+"""
+    show([io::IO,] df::AbstractDataFrame;
+         allrows::Bool = !get(io, :limit, false),
+         allcols::Bool = !get(io, :limit, false),
+         allgroups::Bool = !get(io, :limit, false),
+         rowlabel::Symbol = :Row,
+         summary::Bool = true)
+
+Render a data frame to an I/O stream. The specific visual
+representation chosen depends on the width of the display.
+
+- `io::IO`: The I/O stream to which `df` will be printed.
+- `df::AbstractDataFrame`: The data frame to print.
+- `allrows::Bool `: Whether to print all rows, rather than
+  a subset that fits the device height. By default this is the case only if
+  `io` has the `limit` `IOContext` property set.
+- `allcols::Bool`: Whether to print all columns, rather than
+  a subset that fits the device width. By default this is the case only if
+  `io` has the `limit` `IOContext` property set.
+- `allgroups::Bool`: Whether to print all groups rather than
+  the first and last, when `df` is a `GroupedDataFrame`.
+  By default this is the case only if `io` has the `limit` `IOContext` property set.
+- `rowlabel::Symbol = :Row`: The label to use for the column containing row numbers.
+- `summary::Bool = true`: Whether to print a brief string summary of the data frame.
+
+# Examples
+```jldoctest
+julia> using DataFrames
+
+julia> df = DataFrame(A = 1:3, B = ["x", "y", "z"]);
+
+julia> show(df, allcols=true)
+3×2 DataFrame
+│ Row │ A │ B │
+├─────┼───┼───┤
+│ 1   │ 1 │ x │
+│ 2   │ 2 │ y │
+│ 3   │ 3 │ z │
+```
+"""
 function Base.show(io::IO,
-                   df::AbstractDataFrame,
-                   allcols::Bool = false,
+                   df::AbstractDataFrame;
+                   allrows::Bool = !get(io, :limit, false),
```

Comment:
So we do the following:

* if `show` is passed `io` then by default we show all cols and rows;
* if `show` is not passed `io` then by default we do not show all cols and rows (given signature below);

and

* we set `splitchunks` to true only if `io` has `:limit` set to `true` (in particular if `:limit` is missing we do not split chunks).

Just want to make sure this is what you intended.

Reply:
Mmm, indeed the last point isn't right. That means `show(df)` is totally unreadable by default. The fact that `stdout` doesn't have standard `IOContext` properties set is annoying.

So I've pushed a commit adding a `splitchunks` arguments defaulting to `true` for the two-argument method. I've also added a test to ensure we catch regressions.